File: fibonacci/r/code.R
# Recursive Fibonacci function
fibonacci <- function(n) {
  if (n == 0) return(0)
  if (n == 1) return(1)
  return(fibonacci(n - 1) + fibonacci(n - 2))
}

# Parse command-line argument for `u`
args <- commandArgs(trailingOnly = TRUE)
u <- as.numeric(args[1])

# Initialize result
r <- 0

# Sum up Fibonacci values for numbers from 1 to `u`
for (i in 1:(u - 1)) {
  r <- r + fibonacci(i)
}

# Print the result
cat(r, "\n")

File: fibonacci/julia/code.jl
fibonacci(n) =
    n == 0 ? 0 :
    n == 1 ? 1 :
    fibonacci(n-1) + fibonacci(n-2)

function main()
    u = parse(Int, ARGS[1])
    r = 0
    for i in 1:(u-1) 
        r += fibonacci(i)
    end
    println(r)
end

main()
